### DISCUSSION TITLE
$Guests and foreach

### DISCUSSION THREAD

**Discussion by Invertible95** [PowerShell/PowerShell#17265] [Q&A] [answered]
Hello,
I'm having trouble with the following code. Am I in the wrong or does `$item = foreach` not work together?

`[array]$item = foreach($items in $item) {
   Get-AzureADUser | Select Displayname, UserPrincipalName, Mail, ObjectId | Sort DisplayName
   }  `

If I run the `foreach` part seperatly it works just fine, but not when I add `$item`

**Comment by Northman-de** [comment, marked as answer]
The line does not make any sense... :-)
Hints:
- in general it should be $item in $items, not vice versa
- you do not use the foreach-variable in the scriptblock. E.g. `foreach ($u in $users) {get-AzureADUser $u}`
You do not need foreach in this case, as you want to retrieve all users.
- You are using the variable $item in the foreach and as target

Try this:
`[array]$items = Get-AzureADUser | Select Displayname, UserPrincipalName, Mail, ObjectId | Sort DisplayName`

**Reply by Invertible95** [reply]
I tip my hat for you sir!
